utils/arcjet.ts

```
@@ -0,0 +1,321 @@
+        const decision = await chatRateLimitClient.protect(request);
+
+        // Extract rate limiting information from decision
+        const rateLimitResult = decision.results.find(result =>
+            result.reason.isRateLimit?.()
+        );
+
+        let remaining: number | undefined;
+        let resetTime: number | undefined;
+
+        if (rateLimitResult && rateLimitResult.reason.isRateLimit?.()) {
+            const reason = rateLimitResult.reason as any;
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)